internal/command/action_test.go

```diff
@@ -120,6 +128,150 @@ func TestRunAction(t *testing.T) {
 
 		So(string(content), ShouldEqual, expectedValue)
 	})
+
+	Convey("@SUMMONDOCKERARGS and @SUMMONENVFILE", t, func() {
+		var stdBuf bytes.Buffer
+		// Run docker wrapped around summon and leveraging @SUMMONDOCKERARGS
+		err := runAction(&ActionConfig{
+			StdErr: &stdBuf,
+			StdOut: &stdBuf,
+			Provider: "/bin/echo", // Use /bin/echo provider for brevity
+			Args: []string{
+				"/bin/sh",
+				"-c",
+				`
+echo @SUMMONDOCKERARGS;
+cat @SUMMONENVFILE;
+`,
+			},
+			YamlInline: `
+A: A_value
+B: B_value
+`,
+		})
+
+		// Make assertions
+		code, err := returnStatusOfError(err)
+		So(err, ShouldBeNil)
+		So(code, ShouldEqual, 0)
+		So(stdBuf.String(), ShouldEqual, "--env A --env B\nA=A_value\nB=B_value\n")
+	})
+
+	Convey("Docker args correctly injected for top-level command", t, func() {
+		RunDockerArgsTestCase(t, func (dockerDaemonSocket string) []string {
+			return []string{
+				"docker",
+				"-H", dockerDaemonSocket,
+				"run",
+				"--rm", "-d", "@SUMMONDOCKERARGS",
+				"alpine",
+			}
+		})
+	})
+
+	Convey("Docker args correctly injected for nested command", t, func() {
+		RunDockerArgsTestCase(t, func (dockerDaemonSocket string) []string {
+			return []string{
+				"sh",
+				"-c",
+				"docker -H "+dockerDaemonSocket+" run --rm -d @SUMMONDOCKERARGS alpine",
+			}
+		})
+	})
+}
+
+func RunDockerArgsTestCase(
+	t *testing.T,
+	dockerCommandGen func(dockerDaemonSocket string,
+) []string) {
+	// This is a test case for @SUMMONDOCKERARGS. It exercises Docker CLI pointed to a mock
+	// server. It asserts on the request payload received on the container creation
+	// endpoint, the volume mounts and environment variables injected by summon are
+	// expected to be present.
```

Comment:
Super fly by comment, so ignore if this is missing something, but I wouldn't have expected this kind of test here.  I would have thought that all we're doing is a pre-process, where the command is updated so that `@SUMMONDOCKERARGS` is replaced with the `--env A --env B...` start or whatever.  Then after that, there is nothing _new_ to test, because you're just running summon with a regular docker command.